src/serialize/value.rs

```diff
@@ -0,0 +1,1050 @@
+            "datetime" => DATETIME_COLUMNS,
+            "IPv4Address" | "IPv6Address" => IP_COLUMNS,
+            "time" => TIME_COLUMNS,
+            "UUID" => UUID_COLUMNS,
+
+            _ => {
+                return Err(SerializationError::new(
+                    PythonDriverSerializationError::UnknownColumnType(name),
+                ));
+            }
+        };
+
+        Ok(columns)
+    }
+
+    fn mismatched_type_error<T: ?Sized>(&self, typ: &ColumnType) -> SerializationError {
+        let expected = match self.map_type_to_expected() {
+            Ok(expected) => expected,
+            Err(e) => return e,
+        };
+        SerializationError::new(mk_typck_err::<T>(
+            typ,
+            BuiltinTypeCheckErrorKind::MismatchedType { expected },
+        ))
+    }
+
+    fn serialize_natives<'b>(
+        &self,
+        typ: &ColumnType,
+        native_type: &NativeType,
+        cell_writer: CellWriter<'b>,
+    ) -> Result<WrittenCellProof<'b>, SerializationError> {
+        match native_type {
+            // Integer types.
+            NativeType::TinyInt => self.serialize_native::<i8>(typ, cell_writer),
+            NativeType::SmallInt => self.serialize_native::<i16>(typ, cell_writer),
+            NativeType::Int => self.serialize_native::<i32>(typ, cell_writer),
+            NativeType::BigInt => self.serialize_native::<i64>(typ, cell_writer),
+            NativeType::Counter => {
+                let value = self
+                    .extract::<i64>()
+                    .map_err(|_| self.mismatched_type_error::<Counter>(typ))?;
+
+                let counter = Counter(value);
+                counter.serialize(typ, cell_writer)
+            }
+            NativeType::Varint => {
+                // Number of bits necessary to represent an integer in binary,
+                // EXCLUDING the sign and leading zeros.
+                let bits_len = self
+                    .call_method0("bit_length")
+                    .and_then(|x| x.extract::<usize>())
+                    .map_err(|_| self.mismatched_type_error::<CqlVarint>(typ))?;
+
+                // The `to_bytes()` method requires the number of bytes necessary to represent
+                // the given integer. The '+1' accounts for values such as 128, where `bits_len / 8 = 1`,
+                // but two's-complement representation in big-endian order (CqlVarintBorrowed format)
+                // requires 2 bytes, since a single signed byte covers only the range -128..127.
+                let bytes_len = bits_len / 8 + 1;
+
+                // TODO: I think there might be a way to create PyDict only once
+                // and reuse it in every `Varint` serialization.
+                let py_dict = PyDict::new(self.py());
+                py_dict.set_item("signed", true).map_err(|e| {
+                    SerializationError::new(PythonDriverSerializationError::PythonError(e))
+                })?;
```

Comment:
:thought_balloon: Good observation - this should be doable. Maybe some kind of a `LazyLock` (or `OnceLock`) would allow for this?